src/frontend/src/lib/components/carousel/Carousel.svelte

```diff
@@ -0,0 +1,235 @@
+	 */
+	let container: HTMLDivElement | undefined;
+	let containerWidth = 0;
+
+	/**
+	 * Computed slider frame element that wraps all slides
+	 */
+	let sliderFrame: HTMLDivElement | undefined;
+
+	/**
+	 * Variables related to the slides
+	 */
+	let slides: Node[];
+	let currentSlide = 0;
+	let totalSlides: number;
+
+	/**
+	 * Autoplay timer
+	 */
+	let autoplayTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * Slide transition timer that needed for last-to-first and first-to-last non-animated transform
+	 */
+	let slideTransformTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * A const to be used for enabling transition animation
+	 */
+	const animationParams = {
+		easing,
+		duration
+	};
+
+	/**
+	 * Initialise all required data and attach/detach events
+	 */
+	onMount(() => {
+		initializeCarousel();
+		initialiseAutoplayTimer();
+
+		// Resize listener to re-calculate slide frame width
+		window.addEventListener('resize', onResize);
+
+		return () => {
+			clearAutoplayTimer();
+			clearSlideTransformTimer();
+
+			window.removeEventListener('resize', onResize);
+		};
+	});
+
+	/**
+	 * Build slider frame and set required variables
+	 */
+	const initializeCarousel = () => {
+		if (isNullish(container)) {
+			return;
+		}
+
+		// Save this data only on mount
+		if (isNullish(slides)) {
+			slides = [...container.children];
+			totalSlides = slides.length;
+		}
+
+		containerWidth = container.offsetWidth;
+
+		sliderFrame = buildCarouselSliderFrame({
+			slides,
+			slideWidth: containerWidth
+		});
+
+		// Clean previous HTML if the frame is being re-built (e.g. in case of window resize)
+		container.innerHTML = '';
+		container.appendChild(sliderFrame);
+
+		goToSlide(currentSlide);
+	};
+
+	/**
+	 * Start autoplay timer
+	 */
+	const initialiseAutoplayTimer = () => {
+		autoplayTimer = setInterval(() => {
+			goToNextSlide();
+		}, autoplay);
+	};
+
+	/**
+	 * Clear autoplay timer
+	 */
+	const clearAutoplayTimer = () => {
+		if (nonNullish(autoplayTimer)) {
+			clearInterval(autoplayTimer);
+			autoplayTimer = undefined;
+		}
+	};
+
+	/**
+	 * Reset autoplay timer
+	 */
+	const resetAutoplayTimer = () => {
+		clearAutoplayTimer();
+		initialiseAutoplayTimer();
+	};
+
+	/**
+	 * Clear slide transform timer
+	 */
+	const clearSlideTransformTimer = () => {
+		if (nonNullish(slideTransformTimer)) {
+			clearInterval(slideTransformTimer);
+			slideTransformTimer = undefined;
+		}
+	};
+
+	/**
+	 * Re-initialize carousel on resize
+	 */
+	const onResize = () => {
+		initializeCarousel();
+	};
+
+	/**
+	 * Switch to the next slide
+	 */
+	const goToNextSlide = () => {
+		const nextCurrentSlide = currentSlide + 1;
+
+		enableTransition({ sliderFrame, ...animationParams });
+		goToSlide(nextCurrentSlide);
+
+		if (nextCurrentSlide >= totalSlides) {
+			// Switch to the first slide without animation
+			slideTransformTimer = setTimeout(() => {
+				disableTransition({ sliderFrame });
+				goToSlide(0);
```

Comment:
Have you thhought to passing the animation as an option of `goToSlide` to avoid having to enable or disable before the going to the slide?

```
goToSlide({ slide: 0, transition: none })
```